### PR TITLE
Ft register drivers and operators

### PR DESCRIPTION
**Pull request by Kalimbak**
What does this PR do?
    This PR consists of the registration of the drivers and operators by the administrator in the Phantom project
Description of tasks covered 
    The administrator should be able to register a new driver or a new operator.
    An email containing a random password should be sent to the provided email.
    The driver or operator should be able to sign in to the app using their email and the provided password.
     Appropriate error messages are displayed on unsuccessful attempts

### CODE REVIEW

File: src/routes/userRoute.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+const router = Router();
+// import Users from "../controllers/usersController";
+import { addUser, allUsers, findOneUser, update, deleteUser} from '../controllers/usersController'
+
+
+
+// const router = express.Router();
+
+// get all users
+router.get("/api/v1/users", allUsers);
```
Comment (tjeaneric) [inline src/routes/userRoute.js]: You are repeating users many times in your routes

**Review by tjeaneric — CHANGES_REQUESTED** [review]

File: src/app.js [hunk 2]
```diff
@@ -1,20 +1,35 @@
...
 import i18next from 'i18next';
 import Backend from 'i18next-fs-backend';
 import middleware from 'i18next-http-middleware';
+import dotenv from "dotenv";
+import bodyParser from "body-parser";
+// import { sequelize } from "../models";
+
+dotenv.config()
+
+import { userRouter } from "./routes/userRoute";
```
Comment (kwizeradev) [inline src/app.js]: We are supposed to use appRoutes like on line 24

File: src/controllers/usersController.js [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+  const password = await bcrypt.hash(userpassword, 10)
+  console.log(password,"hello");
+
+  if (firstname === "" || lastname === "" || email === "" || role === "") {
+    return res.status(500).json({
+      message: req.t("All fields must be completed"),
+    });
+  } else if (!(/^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/.test(email))) {
+    return res.status(400).json({
+      message: 'Invalid email'
```
Comment (kwizeradev) [inline src/controllers/usersController.js]: Translate all your messages

**Review by kwizeradev — CHANGES_REQUESTED** [review]

File: src/controllers/RouteController.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    //                 { origin: req.body.origin },
+    //                 { destination: req.body.destination }
+    //             ]
+    //         }
+    //     });
+    //     if (foundItem) {
+    //         return responseHandler(res, 400, "Conflicts occured with existing route.")
+    //     }
+    // }
+
```
Comment (Angelus123) [inline src/controllers/RouteController.js]: Remove unnecessary so as to make code clean

File: src/server.js [hunk 5]
```diff
@@ -1,12 +1,26 @@
...
+// import prop from './config/config';
+// const currentConfig = prop[process.env.NODE_ENV];
+
+// const { port } = currentConfig;
+
+
+// import config from './config/config';
+// const currentConfig = config[process.env.NODE_ENV];
+
+// import app from './app';
```
Comment (Angelus123) [inline src/server.js]: and also here remove unnecessary comment

File: src/controllers/RouteController.js (hunk 4 above)
Comment (Angelus123) [inline src/controllers/RouteController.js]: Remove these comments

**Review by Angelus123 — CHANGES_REQUESTED** [review]

**Review by Angelus123 — APPROVED** [review]

File: src/app.js [hunk 6]
```diff
@@ -1,23 +1,38 @@
...
-     loadPath: './locales/{{lng}}/translation.json'
-   }
+    fallbackLng: 'en',
+    backend: {
+      loadPath: './locales/{{lng}}/translation.json'
+    }
   })
 
 app.use(middleware.handle(i18next));
+app.use('', appRoutes);
```
Comment (tjeaneric) [inline src/app.js]: Correct your routing

**Review by tjeaneric — CHANGES_REQUESTED** [review]

**Review by gerard1001 — APPROVED** [review]

File: locales/fr/translation.json [hunk 7]
```diff
@@ -1,6 +1,14 @@
 {
-  "welcome_message": "Bienvenue au Dominators-Phantom-API!",
```
Comment (tjeaneric) [inline locales/fr/translation.json:2]: you deleted others work

File: locales/kiny/translation.json [hunk 8]
```diff
@@ -1,6 +1,14 @@
 {
-  "welcome_message": "Murakaze neza kuri Dominators-Phantom-API!",
-  "first_message": "Murakaze neza",
-  "incorrect": "Imeyili cyangwa pasiwadi washyizemo ntago arizo",
```
Comment (tjeaneric) [inline locales/kiny/translation.json:4]: You deleted others' work.

File: locales/kisw/translation.json [hunk 9]
```diff
@@ -1,6 +1,14 @@
 {
-  "welcome_message": "Karibuni sana kwa Dominators-Phantom-API!",
-  "first_message": "Karibuni sana",
-  "incorrect": "barua pepe au nenosiri lisilo sahihi",
```
Comment (tjeaneric) [inline locales/kisw/translation.json:4]: You deleted others' work.

File: src/app.js [hunk 10]
```diff
@@ -1,40 +1,53 @@
...
 import swaggerUi from 'swagger-ui-express';
+import swaggerDocument from './documentation/index';
 import cors from 'cors';
+
+import { appRoutes } from './routes'
 import i18next from 'i18next';
 import Backend from 'i18next-fs-backend';
 import middleware from 'i18next-http-middleware';
-import globalErrorHandler from './controllers/errorController';
-import swaggerDocument from './documentation/index';
```
Comment (tjeaneric) [inline src/app.js]: You deleted others' work.

File: src/database/models/user.js [hunk 11]
```diff
@@ -11,29 +13,15 @@ module.exports = (sequelize, DataTypes) => {
...
     }
   }
-  User.init(
-    {
-      firstName: { type: DataTypes.STRING, allowNull: false },
-      lastName: { type: DataTypes.STRING, allowNull: false },
-      email: {
-        type: DataTypes.STRING,
-        allowNull: false,
-        unique: true,
```
Comment (tjeaneric) [inline src/database/models/user.js]: You deleted others' work.

File: src/routes/index.js [hunk 12]
```diff
@@ -1,8 +1,11 @@
 import express from 'express';
-import userRoutes from './api/userRoutes';
+import { busRoutes } from './busRoutes';
+import { userRouter } from './userRoute';
 
-const routes = express.Router();
+const router = express.Router();
 
-routes.use('/users', userRoutes);
```
Comment (tjeaneric) [inline src/routes/index.js]: You deleted others' work.

**Review by tjeaneric — CHANGES_REQUESTED** [review]

File: src/app.js [hunk 13]
```diff
@@ -19,12 +23,12 @@ i18next
...
-      loadPath: './locales/{{lng}}/translation.json',
-    },
-  });
+      loadPath: './locales/{{lng}}/translation.json'
+    }
+  })
 
 app.use(middleware.handle(i18next));
-
+app.use('', appRoutes);
```
Comment (tjeaneric) [inline src/app.js]: correct your routing

File: src/database/models/user.js [hunk 14]
```diff
@@ -1,5 +1,12 @@
+<<<<<<< HEAD
 import { Model } from 'sequelize';
 
+=======
+'use strict';
+const {
+  Model
+} = require('sequelize');
+>>>>>>> Added login feature, jwt on a succesful login and documentation
```
Comment (tjeaneric) [inline src/database/models/user.js]: Conflicts in your codes

File: src/routes/index.js [hunk 15]
```diff
@@ -1,8 +1,21 @@
 import express from 'express';
+<<<<<<< HEAD
```
Comment (tjeaneric) [inline src/routes/index.js]: conflicts messages

**Review by tjeaneric — CHANGES_REQUESTED** [review]

File: src/routes/api/userRoutes.js [hunk 16]
```diff
@@ -1,8 +1,16 @@
 import express from 'express';
 import authController from '../../controllers/authController';
+import { addUser, allUsers, findOneUser, update, deleteUser} from '../../controllers/usersController'
+
+
 
 const router = express.Router();
 
 router.post('/login', authController.login);
+router.post('/register', addUser)
```
Comment (tjeaneric) [inline src/routes/api/userRoutes.js]: Add authentication

File: src/database/models/user.js [hunk 17]
```diff
@@ -25,7 +25,7 @@ module.exports = (sequelize, DataTypes) => {
       role: { type: DataTypes.STRING, allowNull: false },
       password: {
         type: DataTypes.STRING,
-        allowNull: false,
+        allowNull: true,
```
Comment (tjeaneric) [inline src/database/models/user.js]: Why ?

**Review by tjeaneric — CHANGES_REQUESTED** [review]

File: src/database/models/user.js (hunk 17 above)
Comment (Kalimbak) [reply src/database/models/user.js]: I didn't want the password to be required while creating the driver or operator

**Review by tjeaneric — APPROVED** [review]

**Review by Angelus123 — APPROVED** [review]

**Review by furebo — APPROVED** [review]

**Pull request merged**